libs/api/domains/university-of-iceland/src/lib/universityOfIceland.resolver.ts

```diff
@@ -0,0 +1,73 @@
+import { Args, Query, Resolver } from '@nestjs/graphql'
+
+import { Inject, UseGuards } from '@nestjs/common'
+import format from 'date-fns/format'
+import is from 'date-fns/locale/is'
+
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  UniversityOfIcelandService,
+  NemandiGetLocaleEnum,
+  NemandiFerillFerillGetLocaleEnum,
+  StudentTrackTranscripts,
+} from '@island.is/clients/university-of-iceland'
+import { ApiScope } from '@island.is/auth/scopes'
+import { StudentInfoModel } from './models/studentInfo.model'
+import { StudentInfoDetailModel } from './models/studentInfoDetail.model'
+import { GetStudentInfoDetailInput } from './dto/getStudentInfoDetail.input'
+import { GetStudentInfoInput } from './dto/getStudentInfo.input'
+import { DownloadServiceConfig } from '@island.is/nest/config'
+import type { ConfigType } from '@island.is/nest/config'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
+export class UniversityOfIcelandResolver {
```

Comment:
There is a big project this year to unify university information and applications. I would really like to design just one UI and graphql domain (e.g. "university") which abstracts data from multiple sources/clients, even though to start it's only University of Iceland.

Do you think this schema could facilitate other universities?

Reply:
Yeah definitely, however we would rather get this one merged and then refactor later if that's fine

Reply:
I believe this should with perhaps some minor changes/adjustments be able to handle that. When we talked to the service provider initially about data from other Icelandic universities we were told that most of the others will use the same source and the data structure would be very similar/same, which is this structure. We will then need to make some small adjustments for others.
The UI is definitely ready for multiple university handling, we will expect the same from the gql domain.